### PR TITLE
Adding Overridable Error Responses

### PR DESCRIPTION
**Pull request by ppatriotis**
Added ErrorResponseDictionary to manage overrides of errors in MABI. Adjusted extensions and rest of MABI to follow new convention. Added tests for coverage.

### CODE REVIEW

File: ErrorResponseDictionary.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  /**
+   * Should be in the format (all required):
+   *   static $SAMPLE_ERROR_KEY = array(
+   *     'message' => 'sample error message',
+   *     'httpcode' => '401',
+   *     'code' => 1); // optional
+   */
+  function __construct() {
+    $rClass = new \ReflectionClass(get_called_class());
+    $rProperties = $rClass->getProperties(\ReflectionProperty::IS_STATIC);
```
Comment (conord33) [inline ErrorResponseDictionary.php:20]: So are the static array errors being read in here and added to ErrorRessponse array?

File: tests/AppTestCase.php [hunk 2]
```diff
@@ -83,6 +85,7 @@ public function setUpApp($env = array()) {
     );
 
     $this->app->addDataConnection('default', $this->dataConnectionMock);
+    $this->app->getErrorResponseDictionary()->overrideErrorResponses(new Errors());
```
Comment (conord33) [inline tests/AppTestCase.php:88]: this would be added to the config to use the custom errors?

Comment (ppatriotis) [inline tests/AppTestCase.php:88]: Yes. @ignore can be added to ignore the value.

---

Photis Patriotis
215-480-9759

Comment (conord33) [inline tests/AppTestCase.php:88]: sweet

**Comment by conord33** [comment]
Not really sure how I approve this.  But it looks good to me.

**Pull request merged**